frontend/src/resources/utils/get-cluster.ts

```diff
@@ -876,6 +877,13 @@ export function getDistributionInfo(
         upgradeInfo.desiredChannel = isSelectingChannel
             ? clusterCurator?.spec?.upgrade?.channel
             : upgradeInfo.currentChannel
+
+        upgradeInfo.posthookDidNotRun =
+            !upgradeInfo.isUpgrading &&
+            checkCuratorLatestFailedOperation(CuratorCondition.upgrade, curatorConditions) &&
+            upgradeInfo.posthooks?.hasHooks &&
+            !upgradeInfo.posthooks?.success &&
+            !upgradeInfo.posthooks?.failed
```

Comment:
Consider the following scenario:
- Start with cluster with no automation template
- Attempt an upgrade and it fails for some reason
- Complete the upgrade via OCP so the version now matches desired
- Apply an automation template with upgrade posthooks

I believe we will show upgrade failed because posthooks may not have been run in this case, but the failure refers to a previous upgrade attempt. Unless something clears the failure condition?